vars/processAndRunOsdIntegTest.groovy

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+void call(Map args = [:]) {
+    lib = library(identifier: 'jenkins@main', retriever: legacySCM(scm))
+
+    unstash "integtest-opensearch-dashboards-${BUILD_NUMBER}"
+
+    if (env.platform == 'windows') {
+        echo "On Windows Platform, unstash repository and download the artifacts"
+        echo "Downloading from S3: ${artifactPathOpenSearch}"
+        downloadFromS3(
+                assumedRoleName: 'opensearch-bundle',
+                roleAccountNumberCred: 'jenkins-aws-account-public',
+                downloadPath: "${artifactPathOpenSearch}/",
+                bucketName: "${ARTIFACT_BUCKET_NAME}",
+                localPath: "${WORKSPACE}/artifacts",
+                force: true
+        )
+        sh("cp -a ${WORKSPACE}/artifacts/${artifactPathOpenSearch} ${WORKSPACE}")
+
+        echo "Downloading from S3: ${artifactPath}"
+        downloadFromS3(
+                assumedRoleName: 'opensearch-bundle',
+                roleAccountNumberCred: 'jenkins-aws-account-public',
+                downloadPath: "${artifactPath}/",
+                bucketName: "${ARTIFACT_BUCKET_NAME}",
+                localPath: "${WORKSPACE}/artifacts",
+                force: true
+        )
+        sh("cp -a ${WORKSPACE}/artifacts/${artifactPath} ${WORKSPACE}")
+        sh("rm -rf ${WORKSPACE}/artifacts")
+    }
+    else {
+        echo "Not on Windows, unstash repository+artifacts"
+    }
+
+    sh("rm -rf test-results")
+
+    runIntegTestScript(
+            jobName: "${BUILD_JOB_NAME}",
+            componentName: args.localComponent,
```

Comment:
The lib name says OSD so I think componentName will be static here to `OpenSearch-Dashboards`? Maybe add a check and fail if that's not the case?

Reply:
No, for now OSD will be passed but tomorrow if any other component decides to use ci-groups, this need not change, all will be controlled from calling njenkinsfile.